### PR TITLE
better display of errrors

### PR DESCRIPTION
**Pull request by Joshan90**
Readable errors

created custom class for errors and displayed custom messages which are user friendly
works on file system which informs the user what error it encountered in a very basic understandable way in error code
This helps user to quickly figure out where is the mistake and aids in rectifying the mistake swiftly

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@Joshan90

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Lib/pathlib.py [hunk 1]
```diff
@@ -42,6 +43,28 @@
     1921,  # ERROR_CANT_RESOLVE_FILENAME - fix for broken symlink pointing to itself
 )
 
+#Here we are defining OurCustomError class which is derived from Exception class
+
+class OurCustomError(Exception):
+    """Base class for other exceptions"""
+    pass
+
+class ValueError(OurCustomError):
```
Comment (da-woods) [inline Lib/pathlib.py:52]: These classes are just going to confuse people: they have the same name as the builtin exception classes, but are different. Therefore code that tries to catch `ValueError` will fail to catch `pathlib.ValueError`

File: Lib/pathlib.py [hunk 2]
```diff
@@ -262,7 +285,12 @@ def gethomedir(self, username):
                 drv = ''
             userhome = drv + os.environ['HOMEPATH']
         else:
-            raise RuntimeError("Can't determine home directory")
+          try:
+            raise RuntimeError
+          except RuntimeError:
+            print("Possible solutions please refer")
+            print("For more information Refer /n https://stackoverflow.com/questions/58410380/laravel-to-python-runtimeerror-cant-determine-home-directory")
+            print("Here the error you have encountered as program can't determine home directory")
```
Comment (da-woods) [inline Lib/pathlib.py:293]: 1) it's pretty pointless to raise an exception then immediately catch it 1 line later.
2) The function now continues to run even after the error
3) It's no longer possible for a caller to catch and handle the exceptions. Instead they get the printed error message output whatever they do, but no other indication that the function has failed

**Comment by iritkatriel** [comment]
To add to the comments from @da-woods, pathlib is a library and not an application. It is up to application code to decide how an exception should be handled. The pathlib library raises exceptions so that applications can decide how to handle them.

In general, in order to make a change in the behaviour of python you don't jsut submit a patch, you need to at least create an issue on [bpo](http://bugs.python.org) where it can be discussed. You can also raise such suggestions on the python-ideas mailing list. Large changes require a PEP.

**Pull request closed without merging**